src/inmanta/agent/agent.py

```diff
@@ -334,6 +336,78 @@ def notify(
             self.future.set_result(ResourceActionResult(cancel=False))
 
 
+@dataclasses.dataclass
+class DeployRequest:
+    """
+    A request to perform a deploy
+
+    :param is_repair: is this a full deploy/repair run or incremental?
```

Comment:
The name does not seem to correspond with the docstring.

Reply:
Ah, you mean it's either
- a repair or a full deploy
- an incremental deploy
Is that right? I find the "deploy/repair" part difficult to parse. I read it as either
- a full deploy or a full repair
- an incremental deploy or an incremental repair

Even though that doesn't make sense if you know the fine points of deploys and repairs, when you just read the docstring I think the mistake is easily made. I think renaming it `is_full`/`is_incremental` and rewording the docstring to be less ambiguous would make it all more readable.

Reply:
updated